pkg/service/service_test.go

```diff
@@ -611,7 +615,7 @@ func (m *ManagerTestSuite) TestSyncWithK8sFinished(c *C) {
 
 	// cilium-agent finished the initialization, and thus SyncWithK8sFinished
 	// is called
-	err = m.svc.SyncWithK8sFinished()
+	err = m.svc.SyncWithK8sFinished(func(k8s.ServiceID, *lock.StoppableWaitGroup) bool { return true })
```

Comment:
`SyncWithK8sFinished` is called conditionally (see `initRestore`). Can we add one more test with restore flag disabled, just to make sure we don't have any oops moments?

Reply:
I've made a few attempts here, but I couldn't find any good way to test this condition without a large refactoring. The main issue is that the synchronization logic is triggered asynchronously through a controller, hence making it very difficult to wait for its termination. Given that the `initRestore` function is already untested (hence there's no check that the other operations are also skipped when `RestoreStatus` is unset) and the condition is pretty trivial, I'd personally skip the implementation of this test for the moment.

Reply:
It isn't so much about testing `initRestore`, as about making sure there are no deadlocks, or we don't assume that `syncWithK8sFinished` is always invoked (e.g., standalone LB).

Reply:
Yeah, that logic is already triggered only in case that `RestoreState` is set, and the clientset is enabled (which excludes the standalone LB case). It is currently tricky to further test these aspects, because the `syncWithKsFinished` function (which by itself is covered by unit tests) is run asynchronously. 

I'd propose to defer the introduction of additional tests to a subsequent refactoring which extracts this logic to be executed by an hive job, as mentioned by Chris. I'd not include the refactoring in this PR though, to reduce churn in older versions and because that would complicate backporting given that hive jobs support has only been introduced recently.